src/main/java/com/sourcery/gymapp/backend/workout/model/Workout.java

```diff
@@ -36,4 +43,19 @@ public class Workout extends BaseEntity {
     @ManyToOne
     @JoinColumn(name = "routine_id", referencedColumnName = "id")
     private Routine routine;
+
+    @OneToMany(mappedBy = "workout", cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
+    private List<WorkoutExercise> exercises = new ArrayList<>();
+
+    public void setExercises(List<WorkoutExercise> exercises) {
```

Comment:
also forgot to say, that such logic should be in a service layer. (same with another class that has this setter)